osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs

```diff
@@ -123,6 +123,10 @@ public Drawable GetDrawableComponent(ISkinComponent component)
                                 return SkinUtils.As<TValue>(new BindableFloat(legacy_circle_radius));
 
                             break;
+
+                        case OsuSkinConfiguration.ExpandNumberPiece:
+                            string legacyVersion = source.GetConfig<string, string>("Version")?.Value ?? "1";
```

Comment:
Is this accepted? since this is considered legacy version of skins then this seems a good spot to retrieve the version from,  although it's possible to create `LegacySkinConfiguration` enum but unsure if it's worth it.

Reply:
I'm not sure we want skins to be versioned like this. Would personally prefer a `LegacySkinConfiguration`.

Beyond that, this can't be done everywhere, and I'm not sure it's even correct because `latest` should be `2.7`, and not having a `Version` should default to `1`.

Reply:
I accounted for this carefully, this will return `1` if `Version` doesn't exist in `skin.ini` (null is returned if `Version` key isn't in the `ConfigDictionary`, but if `skin.ini` doesn't exist, this will return `latest` (`DefaultSkinConfiguration` is passed to the legacy skin if `skin.ini` doesn't exist afaik)

Reply:
Wanna double check that? Both of these cases construct a `DefaultSkinConfiguration`:

https://github.com/ppy/osu/blob/cbdf42cd7b9cf782cb0aaa7e88b92ea95f806f67/osu.Game/Skinning/LegacySkin.cs#L37-L41

Fwiw I think we should be working with numeric version numbers all the time. That is having the mapping from latest -> 2.7 and nothing -> 1 explicitly defined rather than making these cases everywhere, because it's going to come up a lot more (spinners for example).